src/stores/app/deploy.js

```diff
@@ -1,255 +1,132 @@
-      if (basic.required && _.isEmpty(basic.default)) {
-        this.checkResult = basic.label;
-        break;
-      }
-      cluster[basic.key] = basic.default;
-    }
-    for (let i = 0; i < this.configNodes.length; i++) {
-      let node = this.configNodes[i];
-      cluster[node.key] = {};
-      for (let j = 0; j < node.properties.length; j++) {
-        let value = node.properties[j].default;
-        if (node.properties[j].required && _.isEmpty(_.toString(node.properties[j].default))) {
-          this.checkResult = node.properties[j].label;
-          break;
-        }
-        if (node.properties[j].type === 'integer') value = parseInt(value);
-        cluster[node.key][node.properties[j].key] = value;
-      }
-    }
-    for (let i = 0; i < this.configEnvs.length; i++) {
-      let temp = this.configEnvs[i];
-      env[temp.key] = {};
-      for (let j = 0; j < temp.properties.length; j++) {
-        if (temp.properties[j].required && _.isEmpty(_.toString(temp.properties[j].default))) {
-          this.checkResult = temp.properties[j].label;
-          break;
-        }
-        env[temp.key][temp.properties[j].key] = temp.properties[j].default;
-      }
-    }
-    if (!this.runtimeId) {
-      this.checkResult = 'Runtime';
-    }
-    this.configData = { cluster, env };
+  yaml2Json = yaml_str => {
+    return flattenObject(yamlJs.safeLoad(yaml_str));
   };
 
   @action
-  fetchVersions = async (params = {}, flag) => {
-    const status = ['draft', 'submitted', 'passed', 'rejected', 'active', 'suspended'];
+  fetchVersions = async (params = {}) => {
+    this.isLoading = true;
     if (!params.status) {
-      params.status = status;
+      params.status = ['draft', 'submitted', 'passed', 'rejected', 'active', 'suspended'];
     }
     params.isGlobalQuery = true;
-
     const result = await this.request.get('app_versions', params);
     this.versions = get(result, 'app_version_set', []);
-    this.versionId = get(this.versions[0], 'version_id');
-    if (flag) {
-      await this.fetchFiles(get(this.versions[0], 'version_id'));
-    }
+    this.isLoading = false;
   };
 
   @action
   fetchRuntimes = async (params = {}) => {
     this.isLoading = true;
     const result = await this.request.get('runtimes', params);
     this.runtimes = get(result, 'runtime_set', []);
-    if (this.runtimes.length > 0) {
-      this.runtimeId = this.runtimes[0].runtime_id;
-      if (!this.isKubernetes) {
-        await this.fetchSubnets(this.runtimeId);
-      }
-    }
     this.isLoading = false;
   };
 
   @action
-  async fetchSubnets(runtimeId) {
-    const result = await this.request.get(`clusters/subnets`, {
+  fetchSubnetsByRuntime = async runtimeId => {
+    this.isLoading = true;
+    const result = await this.request.get(`clusters/subnets`, { 
       runtime_id: runtimeId,
       limit: this.maxLimit
     });
     this.subnets = get(result, 'subnet_set', []);
-    this.subnetId = this.subnets[0] ? this.subnets[0].subnet_id : '';
```

Comment:
No need remove this code. If selected subnet, then change selected runtime, the subnet will show previous value.

Reply:
No, if changed runtime, newly subnets will be fetched, previous subnet won't be set.

Reply:
Previous has a bug, after change runtime, query subnets no data, but the page subnet  label show previous value.